rules-tests/DeadCode/Rector/If_/RemoveUnusedNonEmptyArrayBeforeForeachRector/Fixture/property_foreach.php.inc

```diff
@@ -4,12 +4,7 @@ namespace Rector\Tests\DeadCode\Rector\If_\RemoveUnusedNonEmptyArrayBeforeForeac
 
 final class PropertyForeach
 {
-    private $items = [];
-
-    public function __construct(array $items)
-    {
-        $this->items = $items;
-    }
+    private array $items = [];
 
     public function run()
     {
```

Comment:
IMO this test was wrong because we can't know the type of `$this->items` (it is `mixed` in this case)
-> I simplified the test-case and made sure it works on a native array

Reply:
no, it is clearly set :

- default array on property level
- assigned with always array

the type will always be array. 

Changing to typed property will reduce the functionality of upgrading to better. 

We don't need to ensure the outside object fill usage as that out of scope.

Reply:
We can't know whether the class has some other method which overrides the property with any non-array type - at least the rector rule does not do the necessary checks to verify it , or does it?

Reply:
That seems code smell, and the app already crash from the beginning if not filled with proper data for that truthy check then foreach it, like your other PR, on runtime perspective.

The issue with more and more skipping things like that will reduce the usability of upgrading to better code itself.